spec/models/enterprise_caching_spec.rb

```diff
@@ -66,23 +67,33 @@
         end
 
         context "with an order cycle" do
-          before { oc }
+          before do
+            oc
+            # We need to reload again after the OC touched the enterprise.
+            enterprise.reload
+          end
 
           it "touches enterprise when a classification on that product changes" do
+            pending "product touches distributor on change"
+
             expect {
               classification.save!
               enterprise.reload
             }.to change { enterprise.updated_at }
```

Comment:
Alternatively, we could reload the enterprise before and after:
```suggestion
            }.to change { enterprise.reload.updated_at }
```

I think that this is clearer. That block is executed before and after and then compared. Then it doesn't matter which setup code needed reloading.

My current solution came from the rails7 branch where we don't actually need to reload after create because Rails adjusts the precision of timestamps on creation already. But we still have the potential problem of other setup code touching the enterprise.

So big thank you for your feedback here. I will revise my pull request with this code instead. It will be much easier to understand.